### ISSUE TITLE
SessionSettingsSpec intermittently fails

### ISSUE THREAD

**Issue by SethTisue** [sbt/sbt#4148]
seen at https://scala-ci.typesafe.com/view/scala-2.12.x/job/scala-2.12.x-integrate-community-build/2851/

```
[sbt] [info] SessionSettingsSpec
[sbt] [info] SessionSettings  should
[sbt] [info]   + Be identical for empty map 
[sbt] [error]   x Replace statements 
[sbt] [error]    There is 1 failure
[sbt] [error]    There is 1 failure
[sbt] [error]    List((name := "newName",LineRange(0,1)), (scalaVersion := "2.9.2",LineRange(0,1)), (organization := "jozwikr",LineRange(0,1)), (libraryDependencies := Seq("org.scala-sbt" %% "sbinary" % "0.4.1"),LineRange(1,2)), (lazy val checkPom = taskKey[Unit]("check pom to ensure no <type> sections are generated"),LineRange(3,4)), (checkPom := {
[sbt] [error]    	val pomFile = makePom.value
[sbt] [error]    	val pom = xml.XML.loadFile(pomFile)
[sbt] [error]    	val tpe = pom \\ "type"
[sbt] [error]    	if(!tpe.isEmpty)
[sbt] [error]    		sys.error("Expected no <type> sections, got: " + tpe + " in \n\n" + pom)
[sbt] [error]    },LineRange(3,10)), (scalacOptions := Seq("-deprecation"),LineRange(9,10)), (val b = ( <b/>),LineRange(11,12)), (val a = <aaa>
[sbt] [error]    
[sbt] [error]    </aaa>,LineRange(12,15))) != List((name := "alaMaKota",LineRange(0,1)), (scalaVersion := "2.9.2",LineRange(0,1)), (organization := "jozwikr",LineRange(0,1)), (libraryDependencies := Seq("org.scala-sbt" %% "sbinary" % "0.4.1"),LineRange(1,2)), (lazy val checkPom = taskKey[Unit]("check pom to ensure no <type> sections are generated"),LineRange(3,4)), (checkPom := {
[sbt] [error]    	val pomFile = makePom.value
[sbt] [error]    	val pom = xml.XML.loadFile(pomFile)
[sbt] [error]    	val tpe = pom \\ "type"
[sbt] [error]    	if(!tpe.isEmpty)
[sbt] [error]    		sys.error("Expected no <type> sections, got: " + tpe + " in \n\n" + pom)
[sbt] [error]    },LineRange(3,10)), (scalacOptions := Seq("-deprecation"),LineRange(9,10)), (val b = ( <b/>),LineRange(11,12)), (val a = <aaa>
[sbt] [error]    
[sbt] [error]    </aaa>,LineRange(12,15))) (file:1)
[sbt] [error] Added (1)
[sbt] [error] (name := "newName",LineRange(0,1))
[sbt] [error] 
[sbt] [error] Missing (1)
[sbt] [error] (name := "alaMaKota",LineRange(0,1))
[sbt] [error] 
[sbt] [info] Total for specification SessionSettingsSpec
[sbt] [info] Finished in 435 ms
[sbt] [info] 2 examples, 1 failure, 0 error
```

but then green again in 2852 with the same SHAs, so apparently intermittent

**Comment by eed3si9n** [comment]
Sorry about that. I'd be happy to comment them out until we figure out what's happening.

**Comment by SethTisue** [comment]
some projects have provided a flag we can set to disable known-flaky tests, or have segregated flaky tests in a separate project that we can disable

**Comment by SethTisue** [comment]
seen again today

**Comment by dwijnand** [comment]
There's a reference to `SessionSettingsSpec` in https://github.com/sbt/sbt/pull/3802.

However that landed in 1.1.0 (RC2) and the 2.12 community build is testing sbt 1.1.4 (as of 15 days ago).

So 🤷

**Comment by dwijnand** [comment]
Fixed in https://github.com/sbt/sbt/pull/4175.

**Comment by SethTisue** [comment]
@dwijnand seen again at https://scala-ci.typesafe.com/view/scala-2.12.x/job/scala-2.12.x-integrate-community-build/3035/consoleFull (even though we are using `v1.1.6` tag)

**Comment by SethTisue** [comment]
fwiw, I now know how to disable tests on a per-file basis, with e.g. `"set excludeFilter in (Test, unmanagedSources) := HiddenFileFilter || \"SessionSettingsSpec.scala\""`, so you needn't prioritize this because of the community build, only prioritize it if you're concerned yourselves

**Comment by cunei** [comment]
That was fixed in f2a7e1f1c3911fc9fa911d3fc76f61de28fd7a52, which was merged into 1.1.x on May 29. However, v1.1.6 points to 7e8e18b9fa139354669fe31a96d5d62d9e62188f, which is the commit that just precedes it. The fix is actually not in v1.1.6. @eed3si9n

**Comment by SethTisue** [comment]
okay, the community build config now has a comment about it, for when we move to the next set of tags
